Entities/Meta/CTFMusic.as

```diff
@@ -0,0 +1,289 @@
+{
+	timer++;
+	if (mixer is null || !s_gamemusic)
+		return;
+
+	CRules @rules = getRules();
+	CBlob @blob = getLocalPlayerBlob();
+	if (blob is null)
+	{
+		mixer.FadeOutAll(0.0f, 6.0f);
+		return;
+	}
+
+	CMap@ map = blob.getMap();
+	if (map is null)
+		return;
+
+	if (rules.isWarmup())
+	{
+		gameStarted = false;
+		
+		if (timer % getTicksASecond() * 3 == 0)
+		{
+			Vec2f pos = blob.getPosition();
+
+			bool isUnderground = checkUnderground(pos, map);
+			if (isUnderground)
+			{
+				changeMusic(mixer, world_ambient_underground, 2.0f, 2.0f);
+				toggleAmbience(mixer, false, 1.0f);
+			}
+			else if (pos.y < map.tilemapheight * map.tilesize * 0.2f)
+			{
+				changeMusic(mixer, world_ambient_mountain, 2.0f, 2.0f);
+				toggleAmbience(mixer, false, 1.0f);
+			}
+			else
+			{
+				changeMusic(mixer, world_home, 2.0f, 2.0f);
+				toggleAmbience(mixer, true, 1.0f);
+			}
+		}
+	}
+	else if (rules.isMatchRunning())
+	{
+		if (!gameStarted)
+		{
+			Sound::Play("/fanfare_start.ogg");
+		}
+
+		gameStarted = true;
+		if (timer % getTicksASecond() * (mixer.isPlaying(world_battle) ? 5 : 2) == 0)
+		{
+			Vec2f pos = blob.getPosition();
+
+			GameMusicTag chosen = world_calm;
+
+			// check for ambience -- priority
+			bool isUnderground = checkUnderground(pos, map);
+			if (isUnderground)
+			{
+				chosen = world_ambient_underground;
+				toggleAmbience(mixer, false, 1.0f);
+			}
+			else if (pos.y < map.tilemapheight * map.tilesize * 0.2f)
+			{
+				chosen = world_ambient_mountain;
+				toggleAmbience(mixer, false, 1.0f);
+			}
+			else
+			{
+				toggleAmbience(mixer, true, 1.0f);
+
+				CMap@ map = getMap();
+
+				const f32 mapWidth = map.tilemapwidth * map.tilesize;
+				const f32 teamAreaWidth = mapWidth * 0.3;
+
+				u8 team_num = blob.getTeamNum();
+
+				if (pos.x <= teamAreaWidth) // left side of map
+				{
+					chosen = (team_num == 0 ? world_home : world_battle);
+				}
+				else if (pos.x >= mapWidth - teamAreaWidth) // right side of map
+				{
+					chosen = (team_num == 1 ? world_home : world_battle);
+				}
+				else // mid
+				{
+					chosen = world_calm;
+				}
+			}
+
+			if (chosen != world_battle) // below starts more "heavy" checks -> no need to execute this if we are already on battle theme
+			{
+				CBlob@[] flagBases;
+				if (getBlobsByName("flag_base", @flagBases))
+				{
+					for (uint i = 0; i < flagBases.length; i++)
+					{
+						CBlob @b = flagBases[i];
+						if (b.hasTag("flag missing"))
+						{
+							chosen = world_battle;
+							break;
+						}
+					}
+				}
+
+				CBlob@[] blobsInRadius;
+				if (map.getBlobsInRadius(pos, 48.0f, @blobsInRadius))
+				{
+					for (uint i = 0; i < blobsInRadius.length; i++)
+					{
+						CBlob @b = blobsInRadius[i];
+						if (blobList.find(b.getConfig()) >= 0)
+						{
+
+							if (b.getConfig() == "keg" && !b.hasTag("exploding")) // only exploding kegs
+								continue;
+
+							if (classList.find(b.getConfig() && b.hasTag("dead")) // skip corpses
```

Comment:
This isn't correct I believe you meant classList.find(b.getConfig()) && b.hasTag("dead"). Missing a paranthesis.